File: backend/app/schemas/workers.py
from __future__ import annotations

from datetime import datetime
from typing import Optional

from sqlmodel import SQLModel


class WorkerMemberOut(SQLModel):
  membership_id: str
  user_id: str
  phone: Optional[str] = None
  role_name: str
  joined_at: datetime


class WorkerInviteCreate(SQLModel):
  phone: str
  role_id: str


class WorkerInviteOut(SQLModel):
  invite_id: str
  phone: str
  role_name: str
  expires_at: datetime
  activation_code: Optional[str] = None


class PendingInviteOut(SQLModel):
  invite_id: str
  phone: str
  role_name: str
  created_at: datetime
  expires_at: datetime


class InviteActivateRequest(SQLModel):
  invite_id: str
  code: str
  password: str

File: frontend/types/workers.ts
import { z } from "zod";

export const WorkerMemberSchema = z.object({
  membership_id: z.string(),
  user_id: z.string(),
  phone: z.string().nullable(),
  role_name: z.string(),
  joined_at: z.string(),
});

export const PendingInviteSchema = z.object({
  invite_id: z.string(),
  phone: z.string(),
  role_name: z.string(),
  created_at: z.string(),
  expires_at: z.string(),
});

export const WorkerInviteInputSchema = z.object({
  phone: z.string().min(1),
  role_id: z.string().min(1),
});

export const WorkerInviteResultSchema = z.object({
  invite_id: z.string(),
  phone: z.string(),
  role_name: z.string(),
  expires_at: z.string(),
  activation_code: z.string().optional(),
});

export type WorkerMember = z.infer<typeof WorkerMemberSchema>;
export type PendingInvite = z.infer<typeof PendingInviteSchema>;
export type WorkerInviteInput = z.infer<typeof WorkerInviteInputSchema>;
export type WorkerInviteResult = z.infer<typeof WorkerInviteResultSchema>;
